### PR TITLE
refactor: rely on $config property in ViewDecoratorTrait

### PR DESCRIPTION
**Pull request by mostafakhudair**
Use `View::$config` instead of  `config()` function

### CODE REVIEW

**Review by kenjis — APPROVED** [review]

File: system/View/ViewDecoratorTrait.php [hunk 1]
```diff
@@ -22,9 +21,7 @@ trait ViewDecoratorTrait
      */
     protected function decorateOutput(string $html): string
     {
-        $decorators = \config(ViewConfig::class)->decorators;
-
-        foreach ($decorators as $decorator) {
+        foreach ($this->config->decorators as $decorator) {
```
Comment (MGatner) [inline system/View/ViewDecoratorTrait.php:24]: This is a trait that is not marked internal and does not have a constructor or `$config` property. I'm surprised static analysis allowed this. Safer not to assume the property exists:
```suggestion
        $decorators = $this->config->decorators ?? \config(ViewConfig::class)->decorators;

        foreach ($decorators as $decorator) {
```

**Review by MGatner — CHANGES_REQUESTED** [review]

File: system/View/ViewDecoratorTrait.php (hunk 1 above)
Comment (lonnieezell) [reply system/View/ViewDecoratorTrait.php:24]: Can we commit this and merge? This PR has been hanging out, approved, for a couple of months.

**Review by kenjis — CHANGES_REQUESTED** [review]
Please fix UndefinedClass error.

File: system/View/ViewDecoratorTrait.php [hunk 2]
```diff
@@ -12,7 +12,6 @@
 namespace CodeIgniter\View;
 
 use CodeIgniter\View\Exceptions\ViewException;
-use Config\View as ViewConfig;
```
Comment (MGatner) [inline system/View/ViewDecoratorTrait.php:15]: Need to restore this import.

**Review by MGatner — CHANGES_REQUESTED** [review]

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@mostafakhudair Thank you!

**Pull request merged**